pylanetary/navigation/core.py

```diff
@@ -12,96 +12,112 @@
     
-    #set up quadratic equation for ellipsoid
-    r2 = (req/rpol)**2
-    a = 1 + r2*(np.tan(olrad))**2 #second order
-    b = 2*y1*r2*np.sin(olrad) / (np.cos(olrad)**2) #first order
-    c = x1**2 + r2*y1**2 / (np.cos(olrad))**2 - req**2 #constant
+    img_globe = ccrs.Globe(semimajor_axis=r_e, semiminor_axis=r_p, ellipse=None)
+    source_proj = ccrs.PlateCarree(central_longitude=0, globe=img_globe)  
+    target_proj = ccrs.Orthographic(central_longitude=ob_lon, central_latitude=ob_lat, globe=img_globe) 
 
-    radical = b**2 - 4*a*c
-    #will equal nan outside planet since radical < 0
-    with warnings.catch_warnings():
-        warnings.simplefilter("ignore") #suppresses error for taking sqrt nan
-        x3s1=(-b+np.sqrt(radical))/(2*a)
-        x3s2=(-b-np.sqrt(radical))/(2*a)
-    z3s1=(y1+x3s1*np.sin(olrad))/np.cos(olrad)
-    z3s2=(y1+x3s2*np.sin(olrad))/np.cos(olrad)
-    odotr1=x3s1*np.cos(olrad)+z3s1*np.sin(olrad)
-    odotr2=x3s2*np.cos(olrad)+z3s2*np.sin(olrad)
-    #the two solutions are front and rear intersections with planet
-    #only want front intersection
-    
-    #tricky way of putting all the positive solutions into one array
-    with warnings.catch_warnings():
-        warnings.simplefilter("ignore") #suppresses error for taking < nan
-        odotr2[odotr2 < 0] = np.nan
-        x3s2[odotr2 < 0] = np.nan
-        z3s2[odotr2 < 0] = np.nan
-        odotr1[odotr1 < 0] = odotr2[odotr1 < 0]
-        x3s1[odotr1 < 0] = x3s2[odotr1 < 0]
-        z3s1[odotr1 < 0] = z3s2[odotr1 < 0]
-    
-    odotr,x3,z3 = odotr1,x3s1,z3s1
-    y3 = x1
-    r = np.sqrt(x3**2 + y3**2 + z3**2)
-    
-    #lon_w = np.degrees(np.arctan(y3/x3)) + ob_lon
-    lon_w = np.degrees(np.arctan2(x3,y3)-np.pi/2) + ob_lon 
-    with warnings.catch_warnings():
-        warnings.simplefilter("ignore") #suppresses error for taking < nan
-        lon_w[lon_w < 0] += 360
-        lon_w = lon_w%360
-    lat_c = np.degrees(np.arcsin(z3/r))
-    lat_g = np.degrees(np.arctan(r2*np.tan(np.radians(lat_c))))
+    # Define the input and output grids. source is in lat/lon, target is in km
+    target_ny, target_nx = int(shape[0]), int(shape[1])
+    xx, yy = np.arange(target_nx)-target_nx/2-dx, np.arange(target_ny)-target_ny/2-dy
+    target_x_points, target_y_points = np.meshgrid(xx*pixscale_km, yy*pixscale_km) 
+
+    # Project the longitude and latitude grids
+    source_nx, source_ny = int(360/deg_per_px), int(180/deg_per_px)
+    source_x_coords, source_y_coords, extent = mesh_projection(source_proj, source_nx, source_ny, x_extents=(0,360), y_extents=(-90,90))
+    lon_grid = regrid(source_x_coords, source_x_coords, source_y_coords, source_proj, target_proj, target_x_points,  target_y_points, mask_extrapolated=False)
+    lat_grid_g = regrid(source_y_coords, source_x_coords, source_y_coords, source_proj, target_proj, target_x_points,  target_y_points, mask_extrapolated=False)
 
-    return lat_g, lat_c, lon_w
+    # rotate about new image center
```

Comment:
Rotating at the image center or the sub-observer lon/lat? which is also maybe in the center of the new image?